org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java

```diff
@@ -10023,6 +10024,67 @@ public void test297() {
 	runner.javacTestOptions = JavacTestOptions.Excuse.EclipseWarningConfiguredAsError;
 	runner.runNegativeTest();
 }
+// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2176
```

Comment:
There's not much motivation to put this test into this particular test class, **unless** you also provide a test to show-case the effect of `@SuppressWarnings`, but you don't seem to support this yet, right?

Since javac doesn't seem to report nor provided a `@SW` token for this, we're free to chose any token. 

Or are we sure that suppressing is unneeded? Possible reasons would be:
* reference comparison is intended (e.g., because the client code ensures proper interning)
* the warning is raised erroneously (due to a bug in the implementation)

Either way, I suggest `ProgrammingProblemsTest` as a more suitable home for this test :)

Reply:
I'm looking into `@SuppressWarnings`. Should I just use the `UNLIKELY_ARGUMENT_TYPE`? I've renamed the `ComparingWrapper` to a more descriptive `UnlikelyEqualExpressionArgumentType`. I think that the `@SuppressWarnings("unlikely-arg-type")` would be a good solution.

Reply:
The analogy is compelling. I just wonder if "argument" is a good term in connection to `==`, shouldn't it rather be "operand"? OTOH, we probably don't want to introduce a special new token "unlikely-operand-type" :smile: 

Before we decide this: I asked about `@SuppressWarnings` more out of routine, not necessity. Do you anticipate situations where suppressing this warning is relevant?

Reply:
In highly optimised scenarios where I implement my own cache of Long values or Strings for instance. I tend to use the suppress warnings in "Trust me I know what I'm doing" scenarios, rather than disabling the warning altogether.